### DISCUSSION TITLE
Lunar Calendar Plasma Applet v0.1.1

### DISCUSSION THREAD

**Discussion by BLumia** [BLumia/pineapple-calendar#2] [General]
[Install Guide | 安装指南](https://github.com/BLumia/pineapple-calendar/wiki/%E5%AE%89%E8%A3%85%E6%8C%87%E5%8D%97)

This version is tested under Archlinux and Ubuntu 20.04.

Notice: if you are using Ubuntu **20.04**, you may need to apply a patch to get the calendar day text color correctly. You can manually open `/plasmoid/package/contents/ui/calendar/DayDelegate.qml` and replace all `PlasmaCore.Theme.textColor` with `theme.textColor`.

If you got any issues using the applet, please fire a bug or [create a discussion thread](https://github.com/BLumia/pineapple-calendar/discussions).

<hr /><em>This discussion was created from the release <a href='https://github.com/BLumia/pineapple-calendar/releases/tag/0.1.1'>Lunar Calendar Plasma Applet v0.1.1</a>.</em>

**Comment by mraandtux** [comment]
Ubuntu 20.04系统，KDE桌面版本5.18，已安装icu-devtools, libicu-dev, cmake, extra-cmake-modules, qtdeclarative5-dev, libkf5plasma-dev, libkf5declarative-dev等包（[参见Latte Dock的编译包](https://github.com/KDE/latte-dock/blob/master/INSTALLATION.md)），成功编译安装至用户目录和/usr/，但组件显示出现下列错误：

`装入 QML 文件出错：Type CalendarView unavailable`
`module "QtQml" version 2.15 is not installed`

注：已安装qml-module-qtqml-models2，plasmoidviewer输出如下：

```QML debugging is enabled. Only use this in a safe environment.
Icon theme "gnome" not found.
qml: View QML loaded
org.kde.plasmaquick: Applet preload policy set to 1
file:///usr/share/plasma/wallpapers/org.kde.image/contents/ui/main.qml:76:9: Unable to assign [undefined] to QStringList
file:///usr/share/plasma/wallpapers/org.kde.image/contents/ui/main.qml:75:9: Unable to assign [undefined] to int
file:///usr/share/plasma/wallpapers/org.kde.image/contents/ui/main.qml:75:9: Unable to assign [undefined] to int
file:///usr/share/plasma/wallpapers/org.kde.image/contents/ui/main.qml:76:9: Unable to assign [undefined] to QStringList
Toolbox not loading, toolbox package is either invalid or disabled.
qml: New Containment: ContainmentInterface(0x555d18bf89e0)
file:///usr/share/plasma/shells/org.kde.plasma.plasmoidviewershell/contents/applet/AppletError.qml:33:5: QML IconItem: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
file:///usr/share/plasma/shells/org.kde.plasma.plasmoidviewershell/contents/applet/AppletError.qml:33:5: QML IconItem: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
trying to show an empty dialog
file:///usr/share/plasma/shells/org.kde.plasma.plasmoidviewershell/contents/applet/AppletError.qml:33:5: QML IconItem: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
trying to show an empty dialog
trying to show an empty dialog

**Reply by BLumia** [reply]
你尝试的是 0.1.1 版还是 master HEAD（当前 907637c）版？

**Reply by mraandtux** [reply]
当前版，等下退0.1.1看看

**Reply by BLumia** [reply]
嗯，0.1.1 是测试过只要做最上面提到的修改就可以的（即便不改也只是有颜色不对的问题）。master HEAD 是给最新的 plasma 桌面环境用的，有的组件在旧版里可能没有。

**Reply by mraandtux** [reply]
退到0.1.1并修改1楼的参数后，装到/usr/成功，但装到用户目录/.local/就出问题了。

`装入 QML 文件出错：file:///home/mraandtux/.local/share/plasma/plasmoids/net.blumia.pineapple.calendar/contents/ui/main.qml:15:34: Type FullRepresentation unavailable`
`file:///home/mraandtux/.local/share/plasma/plasmoids/net.blumia.pineapple.calendar/contents/ui/FullRepresentation.qml:19:5: Type LocalCalendar.MonthView unavailable`
`file:///home/mraandtux/.local/share/plasma/plasmoids/net.blumia.pineapple.calendar/contents/ui/calendar/MonthView.qml:222:22: Type DaysCalendar unavailable`
`file:///home/mraandtux/.local/share/plasma/plasmoids/net.blumia.pineapple.calendar/contents/ui/calendar/DaysCalendar.qml:317:13: Type DayDelegate unavailable`
`file:///home/mraandtux/.local/share/plasma/plasmoids/net.blumia.pineapple.calendar/contents/ui/calendar/DayDelegate.qml:14:1: module "net.blumia.calendar" is not installed`

于是打包libcalendarutils，再把编译好的Plasmoid一起放进zip包：
[点此下载pineapple-calendar-0.1.1-ubuntu.zip](https://github.com/BLumia/pineapple-calendar/files/6832582/pineapple-calendar-0.1.1-ubuntu.zip)

**Reply by BLumia** [reply]
感谢，我也在这个 Repo 的 [安装指南](https://github.com/BLumia/pineapple-calendar/wiki/%E5%AE%89%E8%A3%85%E6%8C%87%E5%8D%97) Wiki 页面添加了指向上面这个回复的链接来方便其它用户找到你的预编译软件包 :)

**Comment by mraandtux** [comment]
更新：包名更改为plasma-addons-pineapple-calendar，无需分开安装。（压缩包内有deb，[Ubuntu中文论坛也提供.deb下载](https://forum.ubuntu.com.cn/viewtopic.php?f=95&t=492311)）

[plasma-addons-pineapple-calendar.zip](https://github.com/BLumia/pineapple-calendar/files/7077321/plasma-addons-pineapple-calendar.zip)
